### PR TITLE
fix(extra-filters): add logic for identifying applied extra filters

### PR DESCRIPTION
**Pull request by villebro**
### SUMMARY
Adds support for `applied_filters` and `rejected_filters` to chart data endpoint + adds support for temporal extras to both `viz.py` and chart data endpoint.

Short summary of changes:
- In the legacy data `json_explore` endpoint, temporal extras (`__time_range` etc) are applied to `form_data` in `merge_extras` (see: https://github.com/apache/incubator-superset/blob/735123d1f5513d366955db1865faf6b064c9bd4a/superset/utils/core.py#L922-L953 ), information of which temporal filters have been overridden is lost. To get around this, we add the applied temporal extras in a dict called `applied_time_extras` that we can later reference to determine which of them were applied and which were not.
- since `superset-ui/core` merges temporal extras to the query object prior to sending the request, essentially performing the above operation (see: https://github.com/apache-superset/superset-ui/blob/master/packages/superset-ui-core/src/query/extractExtras.ts) , we need to pass along `applied_time_extras` in the request.

**Please note**: needs to upgrade to new version of `superset-ui/core` introduced here to function properly with v1 API plugins: https://github.com/apache-superset/superset-ui/pull/809 .

### SCREENSHOTS

In the below screenshot we can see that all temporal SQL filters are showing up as applied, while the druid ones aren't, as they aren't applicable to SQL datasources:
image

In the below screenshot we can see that all filters are flagged as incompatible, as the datasource doesn't have the gender column, nor any temporal SQL or Druid columns:
image

In this screenshot, we have both legacy (top row) and v1 (bottom row) viz plugins on the same dashboard. Notice that the filters apply similar to the legacy plugins:
image

### TEST PLAN
CI with updated old tests + new E2E test

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/charts/schemas.py [hunk 1]
```diff
@@ -705,6 +705,11 @@ class ChartDataExtrasSchema(Schema):
 
 
 class ChartDataQueryObjectSchema(Schema):
+    applied_time_extras = fields.Dict(
+        description="A mapping of temporal extras that have been applied to the query",
+        required=False,
+        example={"__time_range": "1 year ago : now"},
+    )
```
Comment (villebro) [inline superset/charts/schemas.py:712]: I tried creating a dedicated schema with named properties (`__time_range` etc) and then adding a nested reference here, but that didn't work, probably due to the double underscores. Leaving these as a `fields.Dict` for now to get around the double underscore problem.

File: superset/charts/schemas.py [hunk 2]
```diff
@@ -816,6 +821,13 @@ class ChartDataQueryObjectSchema(Schema):
         "as `having_druid`.",
         deprecated=True,
     )
+    druid_time_origin = fields.String(
+        description="Starting point for time grain counting on legacy Druid "
+        "datasources. Used to change e.g. Monday/Sunday first-day-of-week. "
+        "This field is deprecated and should be passed to `extras` "
+        "as `druid_time_origin`.",
+        allow_none=True,
+    )
```
Comment (villebro) [inline superset/charts/schemas.py:830]: Turns out this was missing from the chart data schema. Also added to deprecated field mapping in `query_object.py`.

File: superset-frontend/spec/javascripts/explore/components/QueryAndSaveBtns_spec.jsx [hunk 3]
```diff
@@ -50,7 +50,7 @@ describe('QueryAndSaveButtons', () => {
 
     it('renders buttons with correct text', () => {
       expect(wrapper.find(Button).contains('Run')).toBe(true);
-      expect(wrapper.find(Button).contains(' Save')).toBe(true);
+      expect(wrapper.find(Button).contains('Save')).toBe(true);
```
Comment (villebro) [inline superset-frontend/spec/javascripts/explore/components/QueryAndSaveBtns_spec.jsx:53]: I'm unsure why GH is showing these lines as changed, as I haven't touched them (the branch is based on `apache:feature/filter-p0` and is up-to-date)

File: superset/charts/schemas.py [hunk 4]
```diff
@@ -894,6 +906,12 @@ class ChartDataResponseResult(Schema):
         description="Amount of rows in result set", allow_none=False,
     )
     data = fields.List(fields.Dict(), description="A list with results")
+    applied_filters = fields.List(
+        fields.Dict(), description="A list with applied filters"
+    )
```
Comment (suddjian) [inline superset/charts/schemas.py:911]: I forgot to add this, thank you!

File: superset/models/dashboard.py [hunk 5]
```diff
@@ -156,7 +156,7 @@ class Dashboard(  # pylint: disable=too-many-instance-attributes
     ]
 
     def __repr__(self) -> str:
-        return f"Dashboard<{self.slug or self.id}>"
+        return f"Dashboard<{self.id or self.slug}>"
```
Comment (suddjian) [inline superset/models/dashboard.py:159]: shouldn't this be the other way round? Is this one of those lines you didn't touch?

Comment (suddjian) [reply superset/models/dashboard.py:159]: I guess if it's for the `__repr__` function it makes more sense to use the id

Comment (villebro) [reply superset/models/dashboard.py:159]: I haven't changed these lines, I think this is GH acting up showing some changes that have come in after the branch was cut.

Comment (suddjian) [reply superset/models/dashboard.py:159]: Huh. Maybe you merged master or something

**Review by suddjian — APPROVED** [review]
Thanks Ville!

**Pull request merged**